core/objects.go

```diff
@@ -146,6 +147,50 @@ type ValidationRecord struct {
 	AddressesTried []net.IP `json:"addressesTried,omitempty"`
 }
 
+// RehydrateHostPort mutates a validation record. If the URL in the validation
+// record cannot be parsed, an error will be returned. If the Hostname and Port
+// fields already exist in the validation record, they will be retained.
+// Otherwise, the Hostname and Port will be derived and set from the URL field
+// of the validation record.
+//
+// CABF BRs section 1.6.1: Authorized Ports: One of the following ports: 80
+// (http), 443 (https)
+func (vr *ValidationRecord) RehydrateHostPort() error {
```

Comment:
Rather than making this a public method on a ValidationRecord, it should be a private/unexported function in the SA which takes a `*core.ValidationRecord` as its only argument. That way it's really clear that all of the hostport manipulation is completely confined to the SA.